src/Orc.DataAccess/Database/DataSourceSchemaProviders/SystemSqlDataSourceSchemaProvider.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Orc.DataAccess.Database;
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+
+[ConnectToProvider("System.Data.SqlClient")]
+public class SystemSqlDataSourceSchemaProvider : IDataSourceSchemaProvider
+{
+    public DbDataSourceSchema? GetSchema(DbConnectionString connectionString)
+    {
+        ArgumentNullException.ThrowIfNull(connectionString);
+
+        var provider = DbProvider.GetRegisteredProvider(connectionString.DbProvider.InvariantName);
+        var databases = new List<string>();
+        using var sqlConnection = provider.CreateConnection();
+        if (sqlConnection is null)
+        {
+            return null;
+        }
+
+        sqlConnection.ConnectionString = connectionString.ToString();
+        sqlConnection.Open();
+
+        using var command = sqlConnection.CreateCommand();
+        command.Connection = sqlConnection;
+        command.CommandText = "SELECT name from sys.databases";
```

Comment:
I think it's best practice to add [sys].[databases]

Reply:
a put brackets as you suggested, even that in this situation it will work without them... but to avoid logic around keywords/spaces etc... is better to have them